### DISCUSSION TITLE
[Feature] Re-match unlabeled faces

### DISCUSSION THREAD

**Discussion by eygraber** [immich-app/immich#5507] [Feature Request]
### The feature

Coming from https://github.com/immich-app/immich/discussions/3812 and with the upcoming feature to unassign faces, I think it would be cool if there was a way to re-match any unassigned faces in case more data is available.

For example, I upload 100 photos of the same person. 80 of them have 4 new faces created for them (20 each). I merge those 80 into one person. The other 20 got incorrectly assigned to another existing person. I then unassign them, and instead of manually merging them into the correct person, I run a rescan of unassigned faces. Hopefully the model is updated enough to assign them to the correct person this time.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by zodcas** [comment]
I read in a previous post on how face recognition works. It uses one picture to gets it algorithm for the face, and then uses that against other detected faces. Having more pictures for a single person does not improve the detection.
I've been playing around with the settings and importing pictures and various things to try and improve things. I have found if I have a folder with specific pictures of each person in them, at different angles/ages/lighting, and importing just those into a fresh immich install, I then get better grouping of pictures.

Even though I can play around in postgres and reassign pictures manually, I'm waiting desperately for 3812!

**Comment by SomeThink1729** [comment]
If I read correctly on the wiki, all points in a cluster that are close enough ( < defined distance, default .5) to enough neighbours are considered a core point.
Having 3 points (with the default settings) located in a triangle all of them are a core point, if I understood correctly, since all of them have enough close neighbours.
Hence having more pictures of the same person should also improve the algorithm, since there are more core points for the person.
Therefore this feature actually would improve the situation after merging many people manually, the clusters around the core points are much more precise so relabeling all unlabeled / unnamed faces would make a difference.
But maybe I'm wrong.

**Comment by ireun** [comment]
Isn't that what a "Missing" job for "Facial recognition" does? I guess this is already implemented?

**Comment by bo0tzz** [comment]
Closing this, as as far as I know you can already do this.
